nifi-nar-bundles/nifi-slack-bundle/nifi-slack-processors/src/main/java/org/apache/nifi/processors/slack/PublishSlack.java

```diff
@@ -0,0 +1,516 @@
+import com.slack.api.methods.response.files.FilesUploadV2Response;
+import com.slack.api.model.Conversation;
+import com.slack.api.model.File;
+import org.apache.nifi.annotation.behavior.DynamicProperty;
+import org.apache.nifi.annotation.behavior.InputRequirement;
+import org.apache.nifi.annotation.behavior.WritesAttribute;
+import org.apache.nifi.annotation.documentation.CapabilityDescription;
+import org.apache.nifi.annotation.documentation.Tags;
+import org.apache.nifi.annotation.lifecycle.OnScheduled;
+import org.apache.nifi.components.AllowableValue;
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.components.ValidationContext;
+import org.apache.nifi.components.ValidationResult;
+import org.apache.nifi.components.Validator;
+import org.apache.nifi.expression.AttributeExpression;
+import org.apache.nifi.expression.ExpressionLanguageScope;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.flowfile.attributes.CoreAttributes;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.processors.slack.consume.ConsumeSlackUtil;
+import org.apache.nifi.processors.slack.publish.ContentMessageStrategy;
+import org.apache.nifi.processors.slack.publish.MessageOnlyStrategy;
+import org.apache.nifi.processors.slack.publish.PublishConfig;
+import org.apache.nifi.processors.slack.publish.PublishSlackClient;
+import org.apache.nifi.processors.slack.publish.PublishStrategy;
+import org.apache.nifi.processors.slack.publish.UploadAttachmentStrategy;
+import org.apache.nifi.processors.slack.publish.UploadLinkStrategy;
+import org.apache.nifi.processors.slack.publish.UploadOnlyStrategy;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import static org.apache.nifi.util.StringUtils.isEmpty;
+
+
+@Tags({"slack", "publish", "notify", "upload", "message"})
+@CapabilityDescription("Uploads the FlowFile content (e.g. an image) as a file, and/or Sends a message on Slack. The FlowFile content can be used as the message text or attached to the message.")
+@InputRequirement(InputRequirement.Requirement.INPUT_REQUIRED)
+@DynamicProperty(name = "<Arbitrary name>", value = "JSON snippet specifying a Slack message \"attachment\"",
+        description = "The property value will be converted to JSON and will be added to the array of attachments in the JSON payload being sent to Slack." +
+                " The property name will not be used by the processor.",
+        expressionLanguageScope = ExpressionLanguageScope.FLOWFILE_ATTRIBUTES)
```

Comment:
I'm not sure that I understand the concept here. As far as I understand it, Slack does not care about the attachment being JSON. Why does the processor care that it's JSON?